model_card_template.md

```diff
@@ -0,0 +1,84 @@
+---
+language:
+- en
+library_name: transformers
+inference: false
+thumbnail: https://h2o.ai/etc.clientlibs/h2o/clientlibs/clientlib-site/resources/images/favicon.ico
+tags:
+- gpt
+- llm
+- large language model
+---
+# Model Card
+## Summary
+
+This model was trained using [H2O LLM Studio](https://github.com/h2oai/h2o-llmstudio).
+- Base model: [{{base_model}}](https://huggingface.co/{{base_model}})
+
+
+## Usage
+
+To use the model with the `transformers` library on a machine with GPUs, first make sure you have the `transformers` and `accelerate` libraries installed.
+
+```bash
+pip install transformers==4.28.1
+pip install accelerate==0.18.0
+```
+
+```python
+import torch
+from transformers import pipeline
+
+generate_text = pipeline(model="{{repo_id}}", torch_dtype=torch.bfloat16, trust_remote_code=True, device_map="auto")
+
+res = generate_text("Why is drinking water so healthy?", max_new_tokens=100)
+print(res[0]["generated_text"])
+```
+
+Alternatively, if you prefer to not use `trust_remote_code=True` you can download [h2oai_pipeline.py](https://huggingface.co/{{repo_id}}/blob/main/h2oai_pipeline.py), store it alongside your notebook, and construct the pipeline yourself from the loaded model and tokenizer:
+
+```python
+import torch
+from h2oai_pipeline import H2OTextGenerationPipeline
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+tokenizer = AutoTokenizer.from_pretrained("{{repo_id}}", padding_side="left")
+model = AutoModelForCausalLM.from_pretrained("{{repo_id}}", torch_dtype=torch.bfloat16, device_map="auto")
+generate_text = H2OTextGenerationPipeline(model=model, tokenizer=tokenizer)
+
+res = generate_text("Why is drinking water so healthy?", max_new_tokens=100)
+print(res[0]["generated_text"])
+```
+
```

Comment:
I think it would still be nice if we clearly state the syntax of the prompt here somewhere. Not everyone will look into the pipeline file to understand it.